src/provider/store/Actions.ts

```diff
@@ -118,4 +118,13 @@ export class ToggleVisibility extends Action<RootState> {
     }
 }
 
+export class ExpireNotification extends Action<RootState> {
```

Comment:
I am not sure if there is any need for this be a store action, as;
it neither has any side-effects on the state nor it has any connection with ui/react (like click events)
Do we consider this as a store action solely because we want this to go through the pipeline and end up in onAction()?

As an alternative you could remove the onAction callback from ExpiryController and expose a method for provider:index to use from its onAction callback and do the APIHandler channel dispatch within that method.

Reply:
Right now the `ExpireNotification` is how `index` know how to fire the `EXPIRE` action. We could dispatch the `EXPIRE` action another way, but this keeps things consistent